Add aten::softmax NNAPI converter

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434
* #59365
* #59364
* #58543
* #60885
* #58541
* #58540
* **#58539**

Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

Commit:
softmax

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "softmax"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "softmax"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

Commit:
Update on "softmax"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "softmax"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).

dreiss commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -1131,6 +1133,27 @@ def add_qadd(self, node, opcode, fuse_code):
+        _, softmax_dim = self.get_constant_value(node.inputsAt(1), "IntType")
+
+        out_id = self.add_tensor_operand(node.outputsAt(0), in_oper)
+        for dim, size in enumerate(in_oper.shape):
+            if size == 0:
+                self.forward_operand_shape(out_id, dim, in_id, dim)
```
Do you think this is a common enough pattern to justify making it a helper function?

axitkhurana replied:
I think so, I'll add a clean task

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

Commit:
Update on "Add aten::softmax NNAPI converter"


Summary:

Add support for aten::softmax op in the NNAPI model converter with
flexible size

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_softmax

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28531946](https://our.internmc.facebook.com/intern/diff/D28531946)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28531946).
